packages/@react-spectrum/s2/src/CoachMark.tsx

```diff
@@ -0,0 +1,530 @@
+
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+import {ActionMenuContext} from './ActionMenu';
+import {
+  DialogTriggerProps as AriaDialogTriggerProps,
+  Popover as AriaPopover,
+  ContextValue,
+  DEFAULT_SLOT,
+  DialogContext,
+  OverlayTriggerStateContext,
+  PopoverContext,
+  PopoverProps,
+  Provider,
+  RootMenuTriggerStateContext,
+  useContextProps
+} from 'react-aria-components';
+import {ButtonContext} from './Button';
+import {Card} from './Card';
+import {CheckboxContext} from './Checkbox';
+import {colorScheme, getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import {ColorSchemeContext} from './Provider';
+import {ContentContext, FooterContext, KeyboardContext, TextContext} from './Content';
+import {
+  createContext,
+  ForwardedRef,
+  forwardRef,
+  ReactNode,
+  useContext,
+  useRef
+} from 'react';
+import {DividerContext} from './Divider';
+import {forwardRefType} from './types';
+import {ImageContext} from './Image';
+import {ImageCoordinator} from './ImageCoordinator';
+import {keyframes, raw} from '../style/style-macro' with {type: 'macro'};
+import {mergeStyles} from '../style/runtime';
+import {PressResponder} from '@react-aria/interactions';
+import {SliderContext} from './Slider';
+import {space, style} from '../style' with {type: 'macro'};
+import {useId, useObjectRef, useOverlayTrigger} from 'react-aria';
+import {useLayoutEffect} from '@react-aria/utils';
+import {useMenuTriggerState} from '@react-stately/menu';
+
+export interface CoachMarkProps extends Omit<PopoverProps, 'children' | 'arrowBoundaryOffset' | 'isKeyboardDismissDisabled' | 'isNonModal'>, StyleProps {
+  /** The children of the coach mark. */
+  children: ReactNode,
+
+  size?: 'S' | 'M' | 'L' | 'XL'
+}
+
+const fadeKeyframes = keyframes(`
```

Comment:
Assuming all of this will change in a future PR, but we shouldn't need these anymore now that popover supports CSS transitions. Popover in S2 was updated in the meantime.

Reply:
not using S2 Popover in followup, see https://github.com/adobe/react-spectrum/pull/8095/files#diff-bc1a3b22ba9ce958f364c9100ab798d2a606a2e24c34e5a9d4a3d45cdcbb2107